okio/src/main/java/okio/Buffer.java

```diff
@@ -1364,32 +1556,55 @@ Segment writableSegment(int minimumCapacity) {
       // Pick the first segment to scan. This is the first segment with offset <= fromIndex.
       s = head;
       if (s == null) {
+        Coverage.visitedge("Buffer::indexOf", "s=null");
         // No segments to scan!
         return -1L;
       } else if (size - fromIndex < fromIndex) {
+        Coverage.visitedge("Buffer::indexOf", "size-fromindex<fromindex");
         // We're scanning in the back half of this buffer. Find the segment starting at the back.
         offset = size;
+        boolean loop1Done = false;
         while (offset > fromIndex) {
+          if(!loop1Done){
+            Coverage.visitedge("Buffer::indexOf", "loop1");
+            loop1Done = true;
+          }
           s = s.prev;
           offset -= (s.limit - s.pos);
         }
       } else {
         // We're scanning in the front half of this buffer. Find the segment starting at the front.
         offset = 0L;
+        boolean loop2Done = false;
         for (long nextOffset; (nextOffset = offset + (s.limit - s.pos)) < fromIndex; ) {
+          if(!loop2Done){
+            Coverage.visitedge("Buffer::indexOf", "loop2");
+            loop2Done = true;
+          }
           s = s.next;
           offset = nextOffset;
         }
       }
     }
 
     // Scan through the segments, searching for b.
+    boolean loop3Done = false;
```

Comment:
Can you briefly explain the purpose of the loop2Done thing?

Reply:
I used this for the previous version, when we were reading, replacing and writing to a file for every "visitedge". It is not needed now.